beginner_source/dist_overview.rst

```diff
@@ -15,207 +12,76 @@ to the technology that can best serve your use case.
+
+
+Applying Parallelism To Scale Your Model
+----------------------------------------
+
+Data Parallelism is a widely adopted single-program multiple-data training paradigm
+where the model is replicated on every process, every model replica computes local gradients for
+a different set of input data samples, gradients are averaged within the data-parallel communicator group before each optimizer step.
 
 PyTorch provides several options for data-parallel training. For applications
 that gradually grow from simple to complex and from prototype to production, the
 common development trajectory would be:
 
 1. Use single-device training if the data and model can fit in one GPU, and
    training speed is not a concern.
-2. Use single-machine multi-GPU
-   `DataParallel <https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html>`__
-   to make use of multiple GPUs on a single machine to speed up training with
-   minimal code changes.
-3. Use single-machine multi-GPU
-   `DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__,
-   if you would like to further speed up training and are willing to write a
-   little more code to set it up.
-4. Use multi-machine `DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
-   and the `launching script <https://github.com/pytorch/examples/blob/master/distributed/ddp/README.md>`__,
-   if the application needs to scale across machine boundaries.
-5. Use multi-GPU `FullyShardedDataParallel <https://pytorch.org/docs/stable/fsdp.html>`__
-   training on a single-machine or multi-machine when the data and model cannot
-   fit on one GPU.
-6. Use `torch.distributed.elastic <https://pytorch.org/docs/stable/distributed.elastic.html>`__
-   to launch distributed training if errors (e.g., out-of-memory) are expected or if
-   resources can join and leave dynamically during training.
-
+2. Use `DistributedDataParallel (DDP) <https://pytorch.org/docs/stable/notes/ddp.html>`__,
+   if your model fits in a single GPU but you want to easily scale up training using multiple GPUs.  Use `torchrun <https://pytorch.org/docs/stable/elastic/run.html>`__,
+   to launch multiple pytorch processes if you are you using more than one node.  See also: `Getting Started with Distributed Data Parallel <../intermediate/ddp_tutorial.html>`__
+3. Use `FullyShardedDataParallel (FSDP) <https://pytorch.org/docs/stable/fsdp.html>`__
+   when your model cannot fit on one GPU.  See also: `Getting Started with FSDP <https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html>`__
+4. When using a large enough cluster that FSDP communications becomes a bottleneck, you can introduce Tensor and Pipeline Parallelism.  See `TorchTitan <https://github.com/pytorch/torchtitan>`__ for an end to end example.
```

Comment:
I wonder if we should split out a model parallelism section given that the beginning of this section only mentioned about "data-parallel training". Or we could augment the first paragraph of this section by adding both Data Parallelism and Model Parallelism, so that it make sense to put this bullet point here.

Reply:
well i renamed the section from 'Data Parallelism' to 'Applying Parallelism To Scale Your Model' to make it more general.